### ISSUE TITLE
Support multiple `Server` instance.

### ISSUE THREAD

**Issue by littlesmilelove** [miragejs/miragejs#309]
I want to separate the mock data by module, but `miragejs` seems only surpport one instance, that means I need to locate all mock routes in one file.

**Comment by samselikoff** [comment]
Yes, unfortunately right now that is true because of how we use Pretender.

The guides show how you can export a function to start your server: https://miragejs.com/quickstarts/react/development#step-2-define-your-server

So you could have multiple servers, but only invoke `startServer()` or `makeServer()` (or whatever you call it) when you need to. You just can't have two servers running at once, but you can define as many as you like.

If you want to shut one down and start another, you just need to call `shutdown()` first:

```js
import { makeOneServer, makeAnotherServer } from './servers'

let serverOne = makeOneServer();
// do some things
serverOne.shutdown()
let serverTwo = makeAnotherServer();
// do some more things to interact with the new server
serverTwo.shutdown()
```

**Comment by littlesmilelove** [comment]
@samselikoff I get it, but in my case, I want to separate my apis by `namespace`, so I can define them in separate files, and I also want to set every `namespace passthrough strategy`, for example:
* api/user namespace : `passthrough` enable
* api/order namespace: `passthrough` disable

So, can you consider to provide a way for use define routes by namespace, and in the server, all the namespace routes compose a complete server routes. I think this way will meet all my needs.

**Comment by samselikoff** [comment]
@littlesmilelove sorry about the delayed response!

Would something like this work?

```js
import { Server } from 'miragejs'
import userRoutes from './user-routes'
import orderRoutes from './order-routes'

new Server({
  routes() {
    userRoutes.call(this)
    orderRoutes.call(this)
  }
})

// user-routes.js
export default function() {
  this.get('/users', schema => schema.users.all())
  // ...more routes

  this.passthrough('/api/user/**')
}

// order-routes.js
export default function() {
  this.post('/orders');
  // ...more routes
}
```

I agree we need a nested/namespaced routing DSL but we don't have one today.

**Comment by littlesmilelove** [comment]
@samselikoff  thank you for your suggestion, this can solve part of the problem. And can you provide your roadmap? I would like to help you achieve this feature together.
